src/step.js

```diff
@@ -71,19 +71,32 @@ class Step {
             return Ok(ret)
         }
 
+        let circularStructureCache = []
```

Comment:
this code has no relation to Step behavior or function. a better way would be refactoring it to a helper function

Reply:
Fixed.